docs/docs/spaces.mdx

```diff
@@ -0,0 +1,237 @@
+work by post-processing intent rankings and entity extractions. If you also need entity
+extraction you need to use the `SpacesCRFEntityExtractor` to make that work with spaces.
+
+Finally, you can see that we are only using the `RulePolicy` here.
+
+After you have created your spaces and made adjustments to the `config.yml`, you can
+train your new assistant using `rasa train -c config.yml` and start it afterward with
+`rasa shell`.
+
+
+## How do spaces work?
+In the following we'll look at how things work under the hood in spaces to give you a
+better understanding of what is happening in case you should need it.
+
+One of the most aspects is that spaces form a hierarchy with two layers. At the top
+is the main space which contains shared primitives that should be usable by all spaces.
+The main space is always active. Anything that is defined in a subspace, can
+only be used by that space and not by other subspaces or the main space.
+
+<img alt="two layer space hierarchy" src={useBaseUrl("/img/spaces_hierarchy.png")} />
+
+### What happens during the assembly?
+
+The most important step during the assembly is the prefixing. During this step every
+intent, entity, slot, action, form, utterance that is defined in a space's domain file
+is prefixed (infixed for utterances) with this space's name. For example, an intent
+`ask_transfer_charge` in the `transfer_money` domain would become
+`transfer_money.ask_transfer_charge` and every reference of this intent would be
+adjusted. The final assistant then works on the prefixed data.
+
+An exception to this is the main space. Anything in the main space and all its
+primitives that are used in other spaces, will not be prefixed.
+
+Another exception are rules. They don't have a name that can be prefixed. Instead,
+we add a condition to each rule that it is only applicable while it's space is active.
+
+### How is space activation tracked?
+
+Space activation is currently tracked through boolean slots that are automatically
+generated during assembly for each space.
+
+### How does filter and rerank work?
+
+The filter and rerank component post-processes the intent ranking of the intent
+classification components. It accesses the tracker and checks which space are active
+or would be activated, in case an entry intent is at the top.
+
+It then removes any intents from the ranking that are not possible. Further it also
+removes any entities that are not possible given the space activation status or
+about to be predicted entry intents.
+
+### How does entity recognition work differently?
+
+Usually, entity recognizers in Rasa only return a single label per token in a message.
+Thus, there is no ranking, that could be post-processed as in the case of intents. We
+have built our `SpacesCRFEntityExtractor` in a way that it creates multiple extractors.
+One for each space. Now, during the post-processing step, we can filter out the
```

Comment:
I think this short sentence "One for each space" could be joined to the previous one